### PR TITLE
Fix documentation for `gensim.corpora`. Partial fix #1671

### LINKED ISSUE #1671: Refactor API reference gensim.corpora
**Check this PR for the soon-to-be-deprecated modules:** #1618 

Documented submodules:

- [ ] `__init__.py`
- [x] `bleicorpus.py`
- [x] `csvcorpus.py`
- [x] `dictionary.py`
- [x] `hashdictionary.py`
- [x] `indexedcorpus.py`
- [x] `lowcorpus.py`
- [x] `malletcorpus.py`
- [x] `mmcorpus.py`
- [ ] ~~`shared_corpus.py`~~
- [x] `svmlightcorpus.py`
- [x] `textcorpus.py`
- [x] `ucicorpus.py`
- [x] `wikicorpus.py`

### PR DESCRIPTION
**Pull request by anotherbugmaster**
Fix #1671 

Docs formally comply with numpy style now but not all type annotations and descriptions are there.

### CODE REVIEW

File: gensim/corpora/bleicorpus.py [hunk 1]
```diff
@@ -35,14 +32,25 @@ class BleiCorpus(IndexedCorpus):
...
         """
         Initialize the corpus from a file.
 
-        `fname_vocab` is the file with vocabulary; if not specified, it defaults to
-        `fname.vocab`.
+        Parameters
+        ----------
+        fname : str
+            Serialized corpus's filename
+        fname_vocab : str or None, optional
```
Comment (menshikh-iv) [inline gensim/corpora/bleicorpus.py]: Need to understand how to:
- Document multiple types of argument (i.e. when the parameter can be type `X` or `Y`)
- Document multiple types for "Return" section
- Correctly specify the parent class (if there are many heirs)

File: gensim/corpora/bleicorpus.py [hunk 2]
```diff
@@ -86,14 +105,26 @@ def line2doc(self, line):
...
         There are actually two files saved: `fname` and `fname.vocab`, where
         `fname.vocab` is the vocabulary file.
 
-        This function is automatically called by `BleiCorpus.serialize`; don't
-        call it directly, call `serialize` instead.
+        Parameters
+        ----------
+        fname : str
+            Filename
+        corpus : iterable
```
Comment (menshikh-iv) [inline gensim/corpora/bleicorpus.py]: iterable of ... ? (here and everywhere)

File: gensim/corpora/bleicorpus.py [hunk 3]
```diff
@@ -121,8 +152,17 @@ def save_corpus(fname, corpus, id2word=None, metadata=False):
...
+        """Return document corresponding to `offset`.
+
+        Parameters
+        ----------
+        offset : int
+            Position of the document in the file
+
+        Returns
+        -------
+        list of (int, float)
```
Comment (menshikh-iv) [inline gensim/corpora/bleicorpus.py]: Missing parameter description (here and everywhere)

File: gensim/corpora/indexedcorpus.py [hunk 4]
```diff
@@ -28,20 +18,39 @@
...
+        Examples
+        --------
         >>> # save corpus in SvmLightCorpus format with an index
-        >>> corpus = [[(1, 0.5)], [(0, 1.0), (1, 2.0)]]
-        >>> gensim.corpora.SvmLightCorpus.serialize('testfile.svmlight', corpus)
-        >>> # load back as a document stream (*not* plain Python list)
-        >>> corpus_with_random_access = gensim.corpora.SvmLightCorpus('tstfile.svmlight')
-        >>> print(corpus_with_random_access[1])
-        [(0, 1.0), (1, 2.0)]
+            >>> corpus = [[(1, 0.5)], [(0, 1.0), (1, 2.0)]]
```
Comment (menshikh-iv) [inline gensim/corpora/indexedcorpus.py]: Examples should be executable and split into 3 sections: imports, data preparation, direct functionality
```
>>> from .. import ...
>>> import ...
>>>
>>> data = ...
>>> makesomething(data)
```

File: gensim/corpora/lowcorpus.py [hunk 5]
```diff
@@ -23,12 +23,22 @@
...
+    """
     return [word for word in utils.to_unicode(s).strip().split(' ') if word]
 
 
 class LowCorpus(IndexedCorpus):
-    """
-    List_Of_Words corpus handles input in GibbsLda++ format.
+    """List_Of_Words corpus handles input in GibbsLda++ format.
 
     Quoting http://gibbslda.sourceforge.net/#3.2_Input_Data_Format::
```
Comment (menshikh-iv) [inline gensim/corpora/lowcorpus.py]: Link in other format

File: gensim/corpora/lowcorpus.py [hunk 6]
```diff
@@ -23,12 +23,22 @@
 
 
 def split_on_space(s):
+    """
+
+    Parameters
+    ----------
+    s :
```
Comment (menshikh-iv) [inline gensim/corpora/lowcorpus.py]: ???? (empty descriptions here and everywhere)

Comment (anotherbugmaster) [reply gensim/corpora/lowcorpus.py]: >Docs formally comply with numpy style now but not all type annotations and descriptions are there.

:)

File: gensim/corpora/bleicorpus.py [hunk 7]
```diff
@@ -28,24 +28,22 @@ class BleiCorpus(IndexedCorpus):
...
         Parameters
         ----------
         fname : str
-            Serialized corpus's filename.
-        fname_vocab : str or None, optional
-            Vocabulary file. If not given, searching for the
-            vocab/vocab.txt file.
+            File path to Serialized corpus.
+        fname_vocab : str, optional
+            Vocabulary file. If `fname_vocab` is None, searching for the vocab.txt or `fname_vocab`.vocab file.
```
Comment (anotherbugmaster) [inline gensim/corpora/bleicorpus.py]: Are you sure it's `fname_vocab`.vocab? `fname_vocab` is none, isn't it?

File: gensim/corpora/bleicorpus.py [hunk 8]
```diff
@@ -108,23 +113,25 @@ def line2doc(self, line):
...
+        -----
+        There are actually two files saved: `fname` and `fname.vocab`, where `fname.vocab` is the vocabulary file.
 
         Parameters
         ----------
         fname : str
-            Filename.
-        corpus : iterable
-            Iterable of documents.
+            Path to output filename.
```
Comment (anotherbugmaster) [inline gensim/corpora/bleicorpus.py]: To output file

File: gensim/corpora/bleicorpus.py [hunk 9]
```diff
@@ -108,23 +113,25 @@ def line2doc(self, line):
...
+        corpus : iterable of iterable of (int, float)
+            Input corpus
```
Comment (anotherbugmaster) [inline gensim/corpora/bleicorpus.py]: Obvious, no additional information provided. There's no need to have descriptions for _all_ arguments. :)

File: gensim/corpora/bleicorpus.py [hunk 10]
```diff
@@ -153,16 +160,18 @@ def save_corpus(fname, corpus, id2word=None, metadata=False):
         return offsets
 
     def docbyoffset(self, offset):
-        """Return document corresponding to `offset`.
+        """Get document corresponding to `offset`,
```
Comment (anotherbugmaster) [inline gensim/corpora/bleicorpus.py]: First line of docstring should always end with a dot.

File: gensim/corpora/bleicorpus.py (hunk 7 above)
Comment (menshikh-iv) [reply gensim/corpora/bleicorpus.py]: Yep

File: gensim/corpora/bleicorpus.py [hunk 11]
```diff
@@ -19,30 +17,39 @@
...
     def __init__(self, fname, fname_vocab=None):
         """
-        Initialize the corpus from a file.
 
-        `fname_vocab` is the file with vocabulary; if not specified, it defaults to
-        `fname.vocab`.
+        Parameters
+        ----------
+        fname : str
+            File path to Serialized corpus.
```
Comment (anotherbugmaster) [inline gensim/corpora/bleicorpus.py]: **Path to corpus** here and in other corpora maybe?

File: gensim/corpora/bleicorpus.py [hunk 12]
```diff
@@ -19,30 +17,39 @@
...
+        fname_vocab : str, optional
+            Vocabulary file. If `fname_vocab` is None, searching for the vocab.txt or `fname_vocab`.vocab file.
```
Comment (anotherbugmaster) [inline gensim/corpora/bleicorpus.py]: Vocabulary file. If `fname_vocab` is None, searching for the vocab.txt or `fname`.vocab file.

File: gensim/corpora/bleicorpus.py [hunk 13]
```diff
@@ -86,14 +111,28 @@ def line2doc(self, line):
...
+        Notes
+        -----
+        There are actually two files saved: `fname` and `fname.vocab`, where `fname.vocab` is the vocabulary file.
+
+        Parameters
+        ----------
+        fname : str
+            Path to output filename.
+        corpus : iterable of iterable of (int, float)
+            Input corpus
```
Comment (anotherbugmaster) [inline gensim/corpora/bleicorpus.py]: Still think that it's not necessary. Also, there's a dot missing at the end of the line.

File: gensim/corpora/bleicorpus.py [hunk 14]
```diff
@@ -121,8 +160,19 @@ def save_corpus(fname, corpus, id2word=None, metadata=False):
         return offsets
 
     def docbyoffset(self, offset):
-        """
-        Return the document stored at file position `offset`.
+        """Get document corresponding to `offset`,
```
Comment (anotherbugmaster) [inline gensim/corpora/bleicorpus.py]: The first line should end with a dot.

File: gensim/corpora/indexedcorpus.py [hunk 15]
```diff
@@ -58,25 +68,38 @@ def __init__(self, fname, index_fname=None):
...
-        structure to file `index_fname` (or `fname`.index is not set).
-
-        This relies on the underlying corpus class `serializer` providing (in
-        addition to standard iteration):
+        """Serialize corpus with offset metadata, allows to use direct indexes after loading.
+
+        Parameters
+        ----------
+        fname : str
+            Path to output filename
```
Comment (anotherbugmaster) [inline gensim/corpora/indexedcorpus.py]: Dots at the end of the line. Did I miss these? O_o

File: gensim/corpora/svmlightcorpus.py [hunk 16]
```diff
@@ -100,16 +120,37 @@ def save_corpus(fname, corpus, id2word=None, labels=False, metadata=False):
...
         """
         with utils.smart_open(self.fname) as f:
             f.seek(offset)
             return self.line2doc(f.readline())[0]
+            # TODO: it brakes if gets None from line2doc
 
     def line2doc(self, line):
-        """
-        Create a document from a single line (string) in SVMlight format
+        """Get a document from a single line in SVMlight format,
```
Comment (anotherbugmaster) [inline gensim/corpora/svmlightcorpus.py]: The first line should end with a dot.

File: gensim/corpora/wikicorpus.py [hunk 17]
```diff
@@ -119,12 +165,24 @@ def remove_markup(text):
 def remove_template(s):
     """Remove template wikimedia markup.
 
-    Return a copy of `s` with all the wikimedia markup template removed. See
-    http://meta.wikimedia.org/wiki/Help:Template for wikimedia templates
-    details.
+    Parameters
+    ----------
+    s : str
+        String containing markup template
```
Comment (anotherbugmaster) [inline gensim/corpora/wikicorpus.py]: A dot at the EOL.

File: gensim/corpora/wikicorpus.py [hunk 18]
```diff
@@ -170,13 +239,26 @@ def remove_file(s):
...
 
-    Return list of tokens as utf8 bytestrings.
+    Parameters
+    ----------
+    content : str
+        String without markup (see :func:`~gensim.corpora.wikicorpus.filter_wiki`).
+    token_min_len : int
+        Minimal token length.
+    token_max_len : int
+        Maximal token length
```
Comment (anotherbugmaster) [inline gensim/corpora/wikicorpus.py]: Dot

File: gensim/corpora/wikicorpus.py [hunk 19]
```diff
@@ -198,10 +292,19 @@ def get_namespace(tag):
...
-    Extract pages from a MediaWiki database dump = open file-like object `f`.
+    """Extract pages from a MediaWiki database dump.
 
-    Return an iterable over (str, str, str) which generates (title, content, pageid) triplets.
+    Parameters
+    ----------
+    f : file
+        File-like object.
+    filter_namespaces : list of str or bool
+         Namespaces that will be extracted
```
Comment (anotherbugmaster) [inline gensim/corpora/wikicorpus.py]: Dot

File: gensim/corpora/wikicorpus.py [hunk 20]
```diff
@@ -336,18 +512,23 @@ def __init__(self, fname, processes=None, lemmatize=utils.has_pattern(), diction
...
+        """Iterate over the dump, yielding list of tokens for each article.
 
-        Only articles of sufficient length are returned (short articles & redirects
-        etc are ignored). This is controlled by `article_min_tokens` on the class instance.
-
-        Note that this iterates over the **texts**; if you want vectors, just use
-        the standard corpus interface instead of this function::
+        Notes
+        -----
+        This iterates over the **texts**. If you want vectors, just use the standard corpus interface
```
Comment (anotherbugmaster) [inline gensim/corpora/wikicorpus.py]: Dot

File: gensim/corpora/bleicorpus.py (hunk 7 above)
Comment (menshikh-iv) [reply gensim/corpora/bleicorpus.py]: Not quite, I added correct description

Comment (anotherbugmaster) [reply gensim/corpora/bleicorpus.py]: Still don't get it. It should be \`fname\`.vocab, \`fname_vocab\` is undefined!

Comment (menshikh-iv) [reply gensim/corpora/bleicorpus.py]: Not quite :) I go through the code with ipdb for this case, this is significantly "wider" that we discuss here (I already fix it).

**Pull request merged**